src/filters/H5Zblosc/src/H5Zblosc.jl

```diff
@@ -91,10 +91,16 @@ function blosc_filter(flags::Cuint, cd_nelmts::Csize_t,
         # uncompressed chunk size but it should not be used in a general
         # cases since other filters in the pipeline can modify the buffer
         # size.
-        outbuf_size, cbytes, blocksize = Blosc.cbuffer_sizes(unsafe_load(buf))
-        outbuf = Libc.malloc(outbuf_size)
-        outbuf == C_NULL && return Csize_t(0)
-        status = Blosc.blosc_decompress(unsafe_load(buf), outbuf, outbuf_size)
+        in = unsafe_load(buf)
+        outbuf_size, cbytes, blocksize = Blosc.cbuffer_sizes(in)
+        # We need to preserve the output of cbuffer_sizes under Julia 1.8 (as of 2021/12/11)
+        # Otherwise their Refs get freed and the buffer becomes corrupted
+        # If this fails, consider copying `in` or wrapping that array.
+        GC.@preserve outbuf_size cbytes blocksize begin
```

Comment:
Without this line, H5Zblosc appears to error with Julia 1.8 in development.

`cbuffer_sizes` calls `blosc_cbuffer_sizes` which returns a size values via pointers / `Refs`.

https://github.com/JuliaIO/Blosc.jl/blob/8ab4e3cf60dd45ee598ad879caf99948bd2a2718/src/Blosc.jl#L129-L137
```julia
function cbuffer_sizes(buf)
    s1 = Ref{Csize_t}()
    s2 = Ref{Csize_t}()
    s3 = Ref{Csize_t}()
    ccall((:blosc_cbuffer_sizes,libblosc), Cvoid,
          (Ptr{UInt8}, Ref{Csize_t}, Ref{Csize_t}, Ref{Csize_t}),
          buf, s1, s2, s3)
    return (s1[], s2[], s3[])
end
```

Somehow it appears that the `Refs` are getting garbage collected and the memory in the actual buffer `in` gets corrupted. This does not make sense to me, but this seems to have worked fine until Julia 1.8, so perhaps there is a bug.

Reply:
This has been resolved in https://github.com/JuliaLang/julia/pull/43408 (pending merge)

Reply:
Oh okay, so this is unrelated to https://github.com/JuliaIO/Blosc.jl/issues/86 or does the fix also require making the changes according to your comment in https://github.com/JuliaIO/Blosc.jl/issues/86 ?

Reply:
Looks like the underlying fix has been merged.  I propose we remove mitigating patch here as it won't be an issue henceforth.

Reply:
You can tag a commit as H5Zblosc_v0.1.0 when you make a release. The main convenience of keeping the versions somewhat aligned is making it easy to know what versions are compatible with what other versions.

Reply:
That is true. OTOH, I think it might not be too difficult to determine if a non-backwards compatible change is made to the filter subpackages. At worst, we could always bump the compatible HDF5 version to the latest one in the filter subpackage  every time a change is made.